File: WebClientIS/WebClientIS/Models/Conselho.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace APIIS.Models
{
    public class Conselho
    {
        public int IdConselho { get; set; }

        public string Nome { get; set; }

        public int IdDist { get; set; }


        public Conselho()
        {
            IdConselho = 0;
            Nome = string.Empty;
            IdDist = 0;
        }

       

        public List<Conselho> GetConselho(string what,int id)
        {
            List<Conselho> lista = new List<Conselho>();

            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Conselho/"+what+"/"+id.ToString());

            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

            HttpResponseMessage response = client.GetAsync("").Result;

            if (response.IsSuccessStatusCode)
            {
                var products = response.Content.ReadAsStringAsync().Result;

                var array = JArray.Parse(products);

                foreach (var obj in array)
                {
                    try
                    {
                        lista.Add(obj.ToObject<Conselho>());
                    }
                    catch (Exception ex)
                    {
                        return new List<Conselho>();
                    }
                }

            }

            return lista;
        }
        public List<Conselho> GetConselhos()
        {
            List<Conselho> lista = new List<Conselho>();

            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Conselho/" );

            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

            HttpResponseMessage response = client.GetAsync("").Result;

            if (response.IsSuccessStatusCode)
            {
                var products = response.Content.ReadAsStringAsync().Result;

                var array = JArray.Parse(products);

                foreach (var obj in array)
                {
                    try
                    {
                        lista.Add(obj.ToObject<Conselho>());
                    }
                    catch (Exception ex)
                    {
                        return new List<Conselho>();
                    }
                }

            }

            return lista;
        }
        public bool CreateConselho(Conselho dis)
        {
            HttpClient client = new HttpClient();

            var values = new Dictionary<string, string>
            {
                { "IdConselho", dis.IdConselho.ToString() },
                { "Nome", dis.Nome },
                { "IdDist", dis.IdDist.ToString() }

            };

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Conselho");

            var jsonString = JsonConvert.SerializeObject(values);
            var content = new StringContent(jsonString, Encoding.UTF8, "application/json");

            var resposta = client.PostAsync(client.BaseAddress.ToString(), content).Result;

            if (resposta.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: MobileClient/sampleproject/lib/Models/Conselho.dart
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert' show json;
import 'package:scoped_model/scoped_model.dart';


class Conselho extends Model{

  int idconselho;
  String nome;
  int iddist;


  Conselho({int idc, String name, int idd})
  {
    this.idconselho = idc;
    this.nome = name;
    this.iddist = idd;
  }

  factory Conselho.fromJson(Map<String,dynamic> json){

    return Conselho(
      idc: json['idConselho'] as int,
      name: json['nome'] as String,
      idd: json['idDist'] as int
          
    );

  }

  Future<List<Conselho>> getconselhos() async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/Conselho"),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<Conselho> listau = new List<Conselho>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(Conselho.fromJson(lista[i]));
  

  return listau;
}

Future<List> getconselho(String what, int id) async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/Conselho/"+what+"/"+id.toString()),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<Conselho> listau = new List<Conselho>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(Conselho.fromJson(lista[i]));
  

  return listau;
}

Future<int> createconselho(Conselho cons) async {

  var url ='http://6f85a52b4f05.ngrok.io/api/Conselho';
  var body = json.encode(<String, String>{
      'IdConselho': cons.idconselho.toString(),
      'Nome':cons.nome,
      'IdDist':cons.iddist.toString()

    });


   http.Response response = await http.post(url,
      headers: {"Content-Type": "application/json"},
      body: body
  );

  return response.statusCode;

}

}